idl/query.idl.hh

```diff
@@ -27,6 +27,7 @@ class qr_partition stub [[writable]] {
 
 class query_result stub [[writable]] {
     utils::chunked_vector<qr_partition> partitions; // in ring order
+    std::optional<full_position> last_position [[version 5.1]];
```

Comment:
What happens on the last page? Do we serialize it anyway? The last page is very common (for short queries that fit in a single page).

Reply:
We try to avoid it. In the single partition query code-path we only serialize this if there is a live querier object that is going to be saved and hence we believe there is more data. I'm not sure I added this condition to the range scan code-path, I have to check.